templates/todo/api/csharp-sql/Program.cs

```diff
@@ -3,7 +3,8 @@
 using SimpleTodo.Api;
 
 var builder = WebApplication.CreateBuilder(args);
-builder.Configuration.AddAzureKeyVault(new Uri(builder.Configuration["AZURE_KEY_VAULT_ENDPOINT"]), new DefaultAzureCredential());
+var credential = new ChainedTokenCredential(new AzureDeveloperCliCredential(), new DefaultAzureCredential());
```

Comment:
Do we have a way to share libs yet?  Thinking that chain building is repeated across all project - maybe we can define in one place. P3

Reply:
If `can't share libs`, consider creating a factory method to produce the `ChainedTokenCredential`.
That's the strategy that I implemented for Java.

Reply:
Let start of by unblocking our use case and we can iterate over time.  Ultimately we want the `AzureDeveloperCliCredential` to make it into the `Azure.Identity` nuget package.  At that point it will also be bundled into the `DefaultAzureCredential` and then we can just revert these changes.  

We _could_ create a new C# Library project with this one file but I don't see that value vs just bundling it up with the app code at the moment.

Reply:
The `ChainedTokenCredential(...)` constructor is basically a factory and accepts `n` number of `TokenCredential` instances.  that will be used in the order that they are registered.  It also doesn't appear to be a credential builder in the .NET version like there is in Java since it really isn't required.